src/util/sss_ini.c

```diff
@@ -180,14 +179,29 @@ static int sss_ini_access_check(struct sss_ini *self)
         flags |= INI_ACCESS_CHECK_UID | INI_ACCESS_CHECK_GID;
     }
 
-    ret = ini_config_access_check(self->file,
-                                  flags,
-                                  geteuid(),
-                                  getegid(),
-                                  S_IRUSR, /* r**------ */
-                                  ALLPERMS & ~(S_IWUSR|S_IXUSR));
+    if (geteuid() == 0)
+        return EOK;
 
-    return ret;
+    /* There is no legit reason to have any write/modify permissions on the config file. */
+    const char *name = ini_config_get_filename(self->file);
+    if (name != NULL) {
+        if (access(name, W_OK) == 0) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "%s is writable to sssd, which it should not be.\n", name);
```

Comment:
This doesn't account for cases when
(1) SSSD_USER support isn't built at all (so 'sssd' runs under 'root' unconditionally, though with dropped caps)
(2) SSSD_USER support is built but for some reason service is configured to be run under 'root'